common/src/main/java/earth/terrarium/hermes/api/defaults/FillAndBorderElement.java

```diff
@@ -0,0 +1,81 @@
+package earth.terrarium.hermes.api.defaults;
+
+import com.teamresourceful.resourcefullib.common.color.Color;
+import com.teamresourceful.resourcefullib.common.color.ConstantColors;
+import earth.terrarium.hermes.api.TagElement;
+import earth.terrarium.hermes.utils.ElementParsingUtils;
+import net.minecraft.client.gui.GuiGraphics;
+
+import java.util.Map;
+import java.util.function.IntFunction;
+
+public abstract class FillAndBorderElement implements TagElement {
+
+    /* For background fills and borders.
+    Though terms are borrowed from CSS's Box Model, does not change Hermes' layout model.
+    Does allow inheriting elements to be set taller and/or wider.
+    The area behind the element, plus any padding, may be filled with background color
+    A border may surround the padding area, with its own width and color. */
+
+    protected int backgroundPadding = 0;
+    protected Color backgroundColor = ConstantColors.gray;
+    protected boolean hasBackground = false;
+    protected int borderWidth = 0;
+    protected Color borderColor = ConstantColors.whitesmoke;
+    protected boolean hasBorder = false;
+    protected int hSpacing; // (background + borderWidth);
+    protected int vSpacing; // 1 or (backgroundPadding + borderWidth)
+
+    protected FillAndBorderElement(Map<String, String> parameters) {
+        if (parameters.containsKey("background")) {
+            this.hasBackground = true;
+            String[] backgroundSpecs = parameters.get("background").split(" ");
+            switch (backgroundSpecs.length) {
+                case 2:
+                    this.backgroundColor = ElementParsingUtils.tryParse(backgroundSpecs[1], Color::parse, backgroundColor);
+                case 1:
+                    this.backgroundPadding = ElementParsingUtils.tryParse(backgroundSpecs[0], Integer::parseInt, backgroundPadding);
+            }
+        }
+        if (parameters.containsKey("border")) {
+            this.hasBorder = true;
+            this.borderWidth = 1; // Set a default value when the tag has a border attribute.
+            String[] borderSpecs = parameters.get("border").split(" ");
+            switch (borderSpecs.length) {
+                case 2:
+                    this.borderColor = ElementParsingUtils.tryParse(borderSpecs[1], Color::parse, borderColor);
+                case 1:
+                    this.borderWidth = ElementParsingUtils.tryParse(borderSpecs[0], Integer::parseInt, borderWidth);
+            }
+        }
+        this.hSpacing = backgroundPadding + borderWidth;
+        this.vSpacing = Math.max(1, (backgroundPadding + borderWidth));
+    }
+
+    public void drawBackground(GuiGraphics graphics, int x, int y, float width, float height) {
+
+        IntFunction<Integer> no00Alpha = (c) -> (c >> 24) != 0 ? c : c + (0xFF << 24);
```

Comment:
can be static, also needs a better name

Reply:
Can change name; can't make static, it seems.

If I change make the function `static`, I get errors about referencing the non-static class attributes.
If I make the class attributes static, _every_ element on the page gets the same background and border.

Reply:
Renamed `drawBackground()` to `drawFillAndBorder()` in be031c2

Reply:
I was talking about the IntFunction as it should be a method not a local variable of a function

Reply:
_ah..._ OK, can do; I had thought of it more as a custom, one-line throw away function for keeping two lines later inside the "drawBackground" (now "drawFillAndBorder") from getting overly long.

Reply:
OK, c4f0e14 makes it now a static method, in the same class, named highPassAlpha()